api/hypershift/v1beta1/azure_types.go

```diff
@@ -1,144 +1,177 @@
 	// HostedCluster.Spec.Platform.Azure.ResourceGroupName.
+	// The image is immutable once set.
 	//
-	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Image is immutable"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="image is immutable"
 	// +kubebuilder:validation:Required
 	Image AzureVMImage `json:"image"`
 
-	// DiskSizeGB is the size in GB to assign to the OS disk
-	// CAPZ default is 30GB, https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b3708019a67ff19407b87d63c402af94ca4246f6/api/v1beta1/types.go#L599
+	// diskSizeGB is the size in GB to assign to the OS disk. This should be between 16 and <TODO find a limit>.
+	// When not set, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is 30.
+	// TODO: Is this GB or GiB?
 	//
-	// +kubebuilder:default:=30
 	// +kubebuilder:validation:Minimum=16
 	// +optional
 	DiskSizeGB int32 `json:"diskSizeGB,omitempty"`
 
-	// DiskStorageAccountType is the disk storage account type to use. Valid values are:
-	// * Standard_LRS: HDD
-	// * StandardSSD_LRS: Standard SSD
-	// * Premium_LRS: Premium SDD
-	// * UltraSSD_LRS: Ultra SDD
-	//
-	// Defaults to Premium_LRS. For more details, visit the Azure documentation:
-	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison
+	// diskStorageAccountType is the disk storage account type to use.
+	// Valid values are Standard, StandardSSD, PremiumSSD and UltraSSD and omitted.
+	// Note that Standard means a HDD.
+	// The disk performance is tied to the disk type, please refer to the Azure documentation for further details
+	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison.
+	// When omitted this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is PremiumSSD.
 	//
-	// +kubebuilder:default:=Premium_LRS
-	// +kubebuilder:validation:Enum=Standard_LRS;StandardSSD_LRS;Premium_LRS;UltraSSD_LRS
+	// +kubebuilder:validation:Enum=Standard;StandardSSD;PremiumSSD;UltraSSD
 	// +optional
+	// TODO: Should all disk options come under a struct rather than be top level like this?
 	DiskStorageAccountType string `json:"diskStorageAccountType,omitempty"`
 
-	// AvailabilityZone is the failure domain identifier where the VM should be attached to. This must not be specified
-	// for clusters in a location that does not support AvailabilityZone.
+	// availabilityZone is the failure domain identifier where the VM should be attached to.
+	// This must not be specified for clusters in a location that does not support AvailabilityZone because... TODO: why?
+	// Example values are TODO
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
```

Comment:
Every string field should answer these questions, patterns, character sets, lengths are all important to think about